File: budget/budget-api/app/revenue/domain/repository.py
from user.domain.user import UserName
from time.domain.date import Date
from revenue.domain.revenue import Revenue, RevenueId


class RevenueRepository:

    def findByDateRange(user_name: UserName, start: Date, end: Date) -> list[Revenue]:
        pass

    def save(budgetRevenue: Revenue) -> Revenue:
        pass

    def update(budgetRevenue: Revenue):
        pass

    def delete(id: RevenueId):
        pass

File: budget/budget-api/domain/budget/revenue/repository.go
package revenue

import (
	"context"

	"github.com/mrflick72/budget/budget-api/domain/time/date"
)

type RevenueRepository interface {
	FindFor(ctx context.Context, revenueId RevenueId) (*Revenue, error)

	FindByDateRange(ctx context.Context, start date.Date, end date.Date) ([]Revenue, error)

	Save(ctx context.Context, revenue *Revenue) error

	Delete(ctx context.Context, revenueId RevenueId) error
}
